### PR TITLE
Query post

### PR DESCRIPTION
**Pull request by tburke**
A long keys option can result in a large URL that errors out with Bad Request. CouchDB allows Query keys to be passed in the body of a POST to handle many keys. http://docs.couchdb.org/en/2.1.1/api/ddoc/views.html#post--db-_design-ddoc-_view-view

### CODE REVIEW

**Comment by flimzy** [comment]
Thanks for the PR.  One question:

Can we be confident that POST is identical to GET, with the exception of the `keys` query parameter, for all use cases?  This is explicitly stated in the documentation in the case of [POST /{db}/_design/{ddoc}/_view/{view}](http://docs.couchdb.org/en/2.1.1/api/ddoc/views.html#post--db-_design-ddoc-_view-view), but not for [POST /{db}/_all_docs](http://docs.couchdb.org/en/2.1.1/api/database/bulk-api.html#post--db-_all_docs).  It's quite possible the docs are just out of sync.

**Comment by tburke** [comment]
I saw _all_docs POST since the 2.0.0 docs: http://docs.couchdb.org/en/2.0.0/api/database/bulk-api.html#post--db-_all_docs
I don't know about pre-2.0

**Comment by flimzy** [comment]
I'm pretty sure that endpoint has been supported for a very long time. My concern is rather, whether the semantics of the endpoint are identical, aside from passing `keys` in the body.  If they are identical, there's probably no reason ever to use GET, and simply always to use the POST version (at least when `keys` is specified).

I vaguely remember wondering about this when I first wrote Kivik, and either determined that they weren't identical, or that I was unsure that they were, which is why I stuck with using GET.

**Comment by flimzy** [comment]
I've done some manual testing against CouchDB 1.6.1, and it appears that it is indeed safe to use POST for all requests, and that only `keys` is treated differently for POST as for GET.

So I think this PR is safe, on principle.  I have a few suggestions for changes, which I'll make inline for clarity.

File: db.go [hunk 1]
```diff
@@ -64,11 +64,22 @@ func optionsToParams(opts ...map[string]interface{}) (url.Values, error) {
 
 // rowsQuery performs a query that returns a rows iterator.
 func (d *db) rowsQuery(ctx context.Context, path string, opts map[string]interface{}) (driver.Rows, error) {
+	// Queries with large key options can be converted to POST.
+	method := kivik.MethodGet
```
Comment (flimzy) [inline db.go]: I think this new block of code should be moved to after the `options, err := optionsToParams(opts)` line, as `optionsToParams` does some data normalization, which allows for non-string inputs (which is probably a minority use-case, but non-string keys are permitted for views, so we should support them).

Then `options` will be of type `url.Values`.  I'd just check the first value, ignoring multiples (this is what CouchDB does anyway).  So:

    Body: ioutil.NopCloser(strings.NewReader(fmt.Sprintf(`{"keys":%s}`, keys[0])))

or similar.

**Comment by flimzy** [comment]
Do you want to update this PR to get it merged?  If not, I'll probably close it, and may produce my own version, with my above comments in mind.

**Pull request closed without merging**